tests/integration/test_jd_coverage_gaps.py

```
@@ -0,0 +1,730 @@
+        from methodologies.johnny_decimal.scanner import FolderScanner
+
+        (tmp_path / "10 Finance").mkdir()
+        (tmp_path / "10 Duplicate").mkdir()
+        scanner = FolderScanner()
+        result = scanner.scan_directory(tmp_path)
+        assert "Existing Johnny Decimal numbers detected" in result.detected_patterns
+
+    def test_scan_deep_hierarchy_warning(self, tmp_path: Path) -> None:
+        from methodologies.johnny_decimal.scanner import FolderScanner
+
+        # Create 7 levels of nesting (depth 0–6) to trigger max_depth > 5 warning
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)